hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -135,10 +142,65 @@ public void resume() {
   @Override
   public BackgroundTaskQueue getTasks() {
     BackgroundTaskQueue queue = new BackgroundTaskQueue();
-    queue.add(new DirectoryDeletingService.DirDeletingTask(this));
+    supplier.reInitItr();
+    for (int i = 0; i < DIR_DELETING_CORE_POOL_SIZE; i++) {
+      queue.add(new DirectoryDeletingService.DirDeletingTask(this));
+    }
     return queue;
   }
 
+  @Override
+  public void shutdown() {
+    super.shutdown();
+    supplier.closeItr();
+  }
+
+  private final class Supplier {
+    TableIterator<String, ? extends KeyValue<String, OmKeyInfo>>
+        deleteTableIterator;
+
+    private Supplier() throws IOException {
+      this.deleteTableIterator =
+          getOzoneManager().getMetadataManager().getDeletedDirTable()
+              .iterator();
+    }
+
+    private synchronized Table.KeyValue<String, OmKeyInfo> getItr() {
+      if (deleteTableIterator != null && !deleteTableIterator.hasNext()) {
+        try {
+          if (getOzoneManager().getMetadataManager().getDeletedDirTable()
```

Comment:
if iterator is closed, will next call for hasNext() have any issue? Can simplify code, if table is not empty, reinit only.
if still it does not have next element, return null.

Reply:
Done